### DISCUSSION TITLE
Read only mode for entities with actions / supported features

### DISCUSSION THREAD

**Discussion by elahd** [home-assistant/architecture#757] [Entity Models]
I'm a main contributor to the [Alarm.com custom integration](https://github.com/uvjustin/alarmdotcom). Alarm.com is a white label provider of burglar alarm and home automation services. This request comes out [this user-posted issue](https://github.com/uvjustin/alarmdotcom/issues/109).

Users may want to connect Home Assistant to Alarm.com using a read-only account. This account can view the state of devices like alarm systems, locks, garage doors, lights, etc., but cannot act on these devices. This use case is useful for users who want to display the status of these devices in Home Assistant but, for various reasons (mostly security), only want to control these devices from within the Alarm.com app.

This use case can apply to any device that offers its own app and is not limited to Alarm.com services.

There are two approaches that can work here:

#### Pros

Uses existing HA architecture.

#### Cons

a. Binary sensors don't currently exist for each device type, so we can't benefit from icons, etc., that have already been selected for these devices.
b. Binary sensors can only have two states whereas controllable devices may have many. For example, an alarm control panel can be disarmed, armed in home mode, armed in away mode, etc. A lock can be locked, unlocked, or jammed.

### 2. Give all controllable devices within Home Assistant a "read only" attribute.

This could be implemented as a dedicated attribute (i.e.: `_attr_read_only`), or by using a more granular approach that builds on `_attr_supported_features`. For instance, for an `AlarmControlPanelEntity`, explicitly returning `AlarmControlPanelEntityFeature.NONE` (i.e.: `0`) for `supported_features` would convey that a device is fully read-only.

#### Pros

This fits in nicely with the existing feature enumeration model.

#### Cons

This may be a breaking change for some integrations. Today, returning `0` for `_attr_supported_features` for an alarm control panel still allows the alarm to be disarmed. Also, the default alarm control panel card will still show the "Arm Home" and "Arm Away" buttons, but pressing the buttons will throw an error in the HA logs.

(As an aside, I think that a disarm feature (i.e.: `AlarmControlPanelEntityFeature.DISARM`) should be added as a requirement for users who can arm, but not disarm, an alarm system.)

Feedback would be appreciated!

**Comment by MartinHjelmare** [comment]
I think we should add a supported feature item for disarm.

If the frontend shows arm buttons for an alarm that doesn't support arming, that sounds like a frontend bug.

**Reply by elahd** [reply]
Awesome, that works for alarm control panels.

Can you confirm how this _should_ work for something like a light entity? Light features are defined through both the LightEntityFeature IntEnum and the ColorMode IntEnum.

ColorMode has the values ColorMode.ONOFF, ColorMode.BRIGHTNESS, ColorMode.COLOR_TEMP, ColorMode.HS, ColorMode.XY, ColorMode.RGB, ColorMode.RGBW, ColorMode.RGBWW, and ColorMode.WHITE. 

LightEntityFeature has the values LightEntityFeature.EFFECT, LightEntityFeature.FLASH, LightEntityFeature.TRANSITION.

If a real-world light supports ColorMode.RGBW and LightEntityFeature.EFFECT, but the user only has read only access, should the following arrangement allow a light to display it's color and on/off status while not allowing the user to set these attributes?

```python
_attr_supported_color_modes = 0
_attr_supported_features = 0

@property
def is_on(self) -> bool | None:
    return something

@property
def rgbw_color(self) -> tuple[int, int, int, int] | None:
    return something

@property
def effect_list(self) -> list[str] | None:
    return something

@property
def effect(self) -> str | None:
    return something
```

That is, will Home Assistant still honor and read from functions for states that are missing from the LightEntityFeature and ColorMode Enums?

**Reply by MartinHjelmare** [reply]
I don't know. In general, a read only feature for an integration should probably be managed by the permission system. It's not really built out and not prioritized at the moment.

**Reply by farmio** [reply]
I guess the frontend would have difficulties to render when `_attr_supported_features` is not set correctly. 

A read-only feature that can be set at runtime could be used for #545 too.
